servers/rendering/shader_compiler.cpp

```diff
@@ -1292,12 +1296,48 @@ String ShaderCompiler::_dump_node_code(const SL::Node *p_node, int p_level, Gene
 							} else {
 								code += node_code;
 							}
-						} else {
-							if (normal_roughness_texture_used && i == 2) {
-								// UV coordinate after using normal roughness texture.
-								node_code = "normal_roughness_uv(" + node_code + ".xy)";
+						} else if (actions.check_multiview_samplers && RS::get_singleton()->is_low_end() && is_texture_func && i == 1) {
+							// Texture function on low end hardware (i.e. OpenGL).
+							// We just need to know if the texture supports multiview.
+
+							StringName texture_uniform;
+							bool correct_texture_uniform = false;
+
+							switch (onode->arguments[i]->type) {
+								case SL::Node::TYPE_VARIABLE: {
+									const SL::VariableNode *varnode = static_cast<const SL::VariableNode *>(onode->arguments[i]);
+									texture_uniform = varnode->name;
+									correct_texture_uniform = true;
+								} break;
+								case SL::Node::TYPE_ARRAY: {
+									const SL::ArrayNode *anode = static_cast<const SL::ArrayNode *>(onode->arguments[i]);
+									texture_uniform = anode->name;
+									correct_texture_uniform = true;
+								} break;
+								default:
+									break;
```

Comment:
Hmmm, I guess this is all duplicated this way.

Sorry to nitpick again. But it looks like it would be better to instead introduce the low-end/heigh-end split at line 1244 (``if (correct_texture_uniform) {``)

So like:
```
if (correct_texture_uniform && !RS::get_singleton()->is_low_end()) {
	// Same as it is now
} else if (correct_texture_uniform && RS::get_singleton()->is_low_end()) {
	if (shader->uniforms.has(texture_uniform)) {
		const ShaderLanguage::ShaderNode::Uniform &u = shader->uniforms[texture_uniform];
		if (u.hint == ShaderLanguage::ShaderNode::Uniform::HINT_SCREEN_TEXTURE) {
			multiview_uv_needed = true;
		} else if (u.hint == ShaderLanguage::ShaderNode::Uniform::HINT_DEPTH_TEXTURE) {
			multiview_uv_needed = true;
		} else if (u.hint == ShaderLanguage::ShaderNode::Uniform::HINT_NORMAL_ROUGHNESS_TEXTURE) {
			multiview_uv_needed = true;
		}
	}
	code += node_code;
} else {
	// Same as it is now
}
```
And 1224 would go from:
```
if (!RS::get_singleton()->is_low_end() && is_texture_func && i == 1) {
```
to: 
```
if (is_texture_func && i == 1) {
```
As you already had it in the last iteration

Reply:
Geeeeeeez.... :)

Just kidding, I was tossing a coin on that one when working on this.